File: 2293.py
n,k = map(int,input().split())
m_type = []
for _ in range(n):
    m_type.append(int(input()))

dp=[0 for _ in range(k+1)]
dp[0]=1
#k원이 되도록 하는 경우의 수
for i in m_type:
    for j in range(1,k+1):
        if j-i>=0:
            dp[j] += dp[j-i]
print(dp[k])

File: BOJ/2293.js
let fs = require('fs');
let input = fs.readFileSync('./test.txt').toString().split('\n');
let [n, k] = input.shift().split(' ').map(Number);
const coins = input.map(Number);
const dp = new Array(k + 1).fill(0);
dp[0] = 1;
for (const coin of coins) {
  for (let j = 1; j < k + 1; j++) {
    if (j - coin >= 0) {
      dp[j] += dp[j - coin];
    }
  }
}
console.log(dp[k]);
